mirrord/agent/src/metrics.rs

```diff
@@ -0,0 +1,189 @@
+use http::StatusCode;
+use prometheus::{register_int_gauge, IntGauge};
+use tokio::net::TcpListener;
+use tokio_util::sync::CancellationToken;
+use tracing::Level;
+
+use crate::error::AgentError;
+
+pub(crate) static CLIENT_COUNT: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_client_count",
+        "amount of connected clients to this mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static DNS_REQUEST_COUNT: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_dns_request_count",
+        "amount of in-progress dns requests in the mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static OPEN_FD_COUNT: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_open_fd_count",
+        "amount of open file descriptors in mirrord-agent file manager"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static MIRROR_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_mirror_port_subscription_count",
+        "amount of mirror port subscriptions in mirror-agent"
+    )
+    .expect("Valid at initialization")
+});
+
+pub(crate) static MIRROR_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_mirror_connection_subscription_count",
+        "amount of connections in mirror mode in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_FILTERED_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_steal_filtered_port_subscription_count",
+        "amount of filtered steal port subscriptions in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_UNFILTERED_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_steal_unfiltered_port_subscription_count",
+        "amount of unfiltered steal port subscriptions in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_FILTERED_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> =
+    LazyLock::new(|| {
+        register_int_gauge!(
+            "mirrord_agent_steal_connection_subscription_count",
+            "amount of filtered connections in steal mode in mirrord-agent"
+        )
+        .expect("Valid at initialization!")
+    });
+
+pub(crate) static STEAL_UNFILTERED_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> =
+    LazyLock::new(|| {
+        register_int_gauge!(
+            "mirrord_agent_steal_connection_subscription_count",
+            "amount of unfiltered connections in steal mode in mirrord-agent"
+        )
+        .expect("Valid at initialization!")
+    });
+
+pub(crate) static HTTP_REQUEST_IN_PROGRESS_COUNT: LazyLock<IntGauge> = LazyLock::new(|| {
```

Comment:
This metric is decremented in a wrong place. I tried to find better places to update it, but oh man. What a mess this is is :sweat_smile: 

Let's just count streaming requests, since this should cover all requests in non-ancient mirrord versions anyway. This can be done easily, because the value of the gauge should always be equal to aggregate size of all `TcpStealerApi::response_body_txs` - update gauge whenever we update the mapping, including `TcpStealerApi` drop

Reply:
Changed.